src/bootstrap/flags.rs

```diff
@@ -443,6 +443,19 @@ Arguments:
             }
         };
 
+        let warnings = match matches.opt_str("warnings").as_ref().map(|v| v.as_str()) {
```

Comment:
Could you add this to its own function + a comment re. what is happening?

Reply:
The code looks pretty obvious to me - what exactly would you want in the comment?

Reply:
Well for one it would be good to know what the role of the `warnings` field is (it just says "true => deny"), why `"deny"` implies `true`, etc. But also it is hard to read code when you just have a big blob inside a function.

Also, why isn't the closure `usage` used here instead of using `std::process::exit`? (Moreover, `std::process` is in scope.)

Also, [I feel obliged to echo @oli-obk here.](https://twitter.com/oli_obk/status/1136582096737185795)

Reply:
I've added some docs to the field (previously `warnings`, now `deny_warnings`).

I disagree that this would be good to pull out into a separate function; I think that wouldn't actually help. This is a single statement, not a group, and this function is the one that deals with processing command line arguments. It's also not a repetition of code, i.e., we're not going to be able to have other code which calls this function.

We're not using `usage` because I wanted to tailor the error message specifically to warnings, e.g., by including the permitted values. It's possibly that it could be made better, but I don't personally want to spend the time figuring out how to integrate with `usage`. This works, is simple, and IMO easier to interpret for end-users and future readers of this code.

Reply:
> I've added some docs to the field (previously `warnings`, now `deny_warnings`).

Thanks!

> I disagree that this would be good to pull out into a separate function; I think that wouldn't actually help. This is a single statement, not a group, and this function is the one that deals with processing command line arguments. It's also not a repetition of code, i.e., we're not going to be able to have other code which calls this function.

I don't think that matters so much. By having large functions you force readers to pay attention to details which may be irrelevant to what the function does at large (in this case parsing all the arguments into flags instead of just dealing with `--warnings`) and it makes skimming the code harder. We don't have to have 5 line functions everywhere, but 395 lines is not readable and this is how

Reply:
I guess I don't care that much. I can do it if it'll unblock this PR. I've never read through this whole function though when trying to edit it, so I don't really see the point, and I continue to think it's actually harmful - makes understanding it harder rather than easier in straight-line code. I would have a different opinion if this function was a mess of if statements that were all nested, but it's not.